### PR TITLE
Issue 1025: Kotlin generator doesn't support inheritance

### PR DESCRIPTION
**Pull request by irishshagua**
### PR checklist
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.3.x`, `4.0.x`. Default: `master`.
  - Can't find a 3.3.x branch, so just using default 🤷‍♂️ 
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@jimschubert  @dr4ke616 

### Description of the PR
Kinda fix for [Issue 1025](https://github.com/OpenAPITools/openapi-generator/issues/1025)
  * allVars is duplicating child preoperties when models are inherited. Filter out these duplicates in the KotlinSpringServerCodegen
  * isInherited property was not being populated in the CodegenModel, re-parse the models in the KotlinSpringServerCodegen class and populate the property here
  * Add template support for Kotlin models which require inheritance from a base class to support oneOf declarations in the api yaml

### CODE REVIEW

**Comment by dr4ke616** [comment]
LGTM: @jimschubert any thoughts?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java [hunk 1]
```diff
@@ -498,4 +502,16 @@ public void execute(Template.Fragment fragment, Writer writer) throws IOExceptio
             writer.write(fragment.execute().replaceAll(from, to));
         }
     }
+
+    // Can't figure out the logic in DefaultCodegen but optional vars are getting duplicated when there's
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java]: Agreed with you it should be done in DefautlCodegen. I'll try to figure out a fix later in DefaultCodegen so as to fix the issue for other generators as well.

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
> Can't find a 3.3.x branch, so just using default 

Current master is 3.3.0 as we merged 3.3.x into master previously to prepare for the release.

**Comment by irishshagua** [comment]
Cheers @wing328, changed the git author now...

**Comment by wing328** [comment]
@irishshagua thanks again for the PR, which is included in the v3.3.0 minor release: https://twitter.com/oas_generator/status/1046941449609068544

**Pull request merged**